### PR TITLE
`IsingYY` differentiability

### PR DESCRIPTION
**Pull request by rmoyard**
**Context:**

This PR makes `IsingYY` differentiable for the different devices Jax, TensorFlow and Autograd.

**Description of the Change:**

1. I added `IsingYY` operation to `jax_ops.py`, `autograd_ops.py`, and `tf_ops.py`.
2. I modified the `DefaultQubitAutograd`, `DefaultQubitJAX`, and `DefaultQubitTF` devices to support `IsingYY`.
3. Tests the gradient for all three devices and for different `diff-method`.

### CODE REVIEW

**Comment by josh146** [comment]
@rmoyard is this review ready? Just checking, as no-one has been tagged for review :)

**Comment by rmoyard** [comment]
@josh146 Yes indeed it is ready! I forgot to tag someone.

File: pennylane/devices/autograd_ops.py [hunk 1]
```diff
@@ -201,6 +202,24 @@ def IsingXX(phi):
     return np.cos(phi / 2) * II - 1j * np.sin(phi / 2) * XX
 
 
+def IsingYY(phi):
+    r"""Ising YY coupling gate
```
Comment (josh146) [inline pennylane/devices/autograd_ops.py:206]:
```suggestion
    r"""Ising YY coupling gate.
```

File: pennylane/ops/qubit.py [hunk 2]
```diff
@@ -1849,6 +1849,7 @@ class IsingYY(Operation):
     num_wires = 2
     par_domain = "R"
     grad_method = "A"
+    generator = [np.array([[0, 0, 0, -1], [0, 0, 1, 0], [0, 1, 0, 0], [-1, 0, 0, 0]]), -1 / 2]
```
Comment (josh146) [inline pennylane/ops/qubit.py:1852]: nice :+1:

File: tests/ops/test_qubit_ops.py [hunk 3]
```diff
@@ -879,6 +912,7 @@ def circuit(phi):
         )
 
         res = qml.grad(circuit)(phi)
+        print(res, expected)
```
Comment (josh146) [inline tests/ops/test_qubit_ops.py:915]:
```suggestion
```
Was this during debugging? 🙂

**Review by josh146 — APPROVED** [review]
Thanks @rmoyard, looks good from my end!

File: tests/ops/test_qubit_ops.py (hunk 3 above)
Comment (rmoyard) [reply tests/ops/test_qubit_ops.py]: Certainly!

**Comment by josh146** [comment]
Hi @rmoyard, is this PR ready to be merged?

**Pull request merged**